runtime/src/multi_sig.rs

```diff
@@ -0,0 +1,413 @@
+            Ok(())
+        }
+
+        /// Remove a signer from the multisig. This must be called by the multisig itself.
+        ///
+        /// # Arguments
+        /// * `multi_sig` - Multisig address.
+        /// * `signer` - Signer to remove.
+        pub fn remove_multi_sig_signer(origin, signer: Signer) -> Result {
+            let sender = ensure_signed(origin)?;
+            ensure!(<MultiSigSignsRequired<T>>::exists(&sender), "Multi sig does not exist");
+            <MultiSigSigners<T>>::insert((sender.clone(), signer), false);
+            Self::deposit_event(RawEvent::MultiSigSignerRemoved(sender, signer));
+            Ok(())
+        }
+
+        /// Change number of sigs required by a multisig. This must be called by the multisig itself.
+        ///
+        /// # Arguments
+        /// * `multi_sig` - Multisig address.
+        /// * `sigs_required` - New number os sigs required.
+        pub fn change_sigs_required(origin, sigs_required: u64) -> Result {
+            let sender = ensure_signed(origin)?;
+            ensure!(<MultiSigSignsRequired<T>>::exists(&sender), "Multi sig does not exist");
+            <MultiSigSignsRequired<T>>::insert(&sender, &sigs_required);
+            Self::deposit_event(RawEvent::MultiSigSignaturesRequiredChanged(sender, sigs_required));
+            Ok(())
+        }
+    }
+}
+
+decl_event!(
+    pub enum Event<T>
+    where
+        AccountId = <T as system::Trait>::AccountId,
+    {
+        /// Event for multi sig creation. (Multisig address, Creator address, Signers(pending approval), Sigs required)
+        MultiSigCreated(AccountId, AccountId, Vec<Signer>, u64),
+        /// Event for adding a proposal (Multisig, proposalid)
+        ProposalAdded(AccountId, u64),
+        /// Emitted when a proposal is executed. (Multisig, proposalid, result)
+        ProposalExecuted(AccountId, u64, bool),
+        /// Signer added (Authorization accepted) (Multisig, signer_added)
+        MultiSigSignerAdded(AccountId, Signer),
+        /// Multi Sig Signer Authorized to be added (Multisig, signer_authorized)
+        MultiSigSignerAuthorized(AccountId, Signer),
+        /// Multi Sig Signer removed (Multisig, signer_removed)
+        MultiSigSignerRemoved(AccountId, Signer),
+        /// Change in signatures required by a multisig (Multisig, new_sigs_required)
+        MultiSigSignaturesRequiredChanged(AccountId, u64),
+    }
+);
+
+impl<T: Trait> Module<T> {
+    /// Creates a new proposal
+    pub fn create_proposal(
+        multi_sig: T::AccountId,
+        proposal: Box<T::Proposal>,
+        sender_signer: Signer,
+    ) -> Result {
+        ensure!(
+            Self::ms_signers((multi_sig.clone(), sender_signer)),
+            "not an signer"
+        );
+        let proposal_id = Self::ms_tx_done(multi_sig.clone());
+        <Proposals<T>>::insert((multi_sig.clone(), proposal_id), proposal);
+        let next_proposal_id: u64 = proposal_id + 1u64;
```

Comment:
I've added overflow check increate wallet function but left it out otherwise because these variables are only incremented by 1. It's not practically possible to overflow 64bit number by incrementing with 1 here.